### PR TITLE
avoid unused result warning on gcc 10

### PR DESCRIPTION
**Pull request by msaf1980**

### CODE REVIEW

File: ctest.h [hunk 1]
```diff
@@ -22,6 +22,12 @@
 #define CTEST_IMPL_FORMAT_PRINTF(a, b)
 #endif
 
+#if defined(__GNUC__) && (__GNUC__ >= 4)
```
Comment (aceckel) [inline ctest.h:25]: Clang also supports this attribute, so how about this?
```suggestion
#if defined(__clang__) || (defined(__GNUC__) && (__GNUC__ >= 4))
```

Comment (aceckel) [reply ctest.h:25]: Alternatively, a more portable solution may be to put the `write` call in an `if` statement with an empty body, maybe with a comment explaining why we don't care if the call fails.

Comment (aceckel) [reply ctest.h:25]: Alternatively, a more portable solution may be to put the `write` call in an `if` statement with an empty body, maybe with a comment explaining why we don't care if the call fails.

Comment (msaf1980) [reply ctest.h]: Wrap write call with if is more simple - no worry about compiler specific.

**Pull request closed without merging**